ambari-server/src/main/java/org/apache/ambari/server/controller/internal/AuthResourceProvider.java

```diff
@@ -0,0 +1,177 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ambari.server.controller.internal;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.ambari.server.controller.AmbariManagementController;
+import org.apache.ambari.server.controller.predicate.EqualsPredicate;
+import org.apache.ambari.server.controller.spi.NoSuchParentResourceException;
+import org.apache.ambari.server.controller.spi.NoSuchResourceException;
+import org.apache.ambari.server.controller.spi.Predicate;
+import org.apache.ambari.server.controller.spi.Request;
+import org.apache.ambari.server.controller.spi.RequestStatus;
+import org.apache.ambari.server.controller.spi.Resource;
+import org.apache.ambari.server.controller.spi.ResourceAlreadyExistsException;
+import org.apache.ambari.server.controller.spi.ResourcePredicateEvaluator;
+import org.apache.ambari.server.controller.spi.ResourceProvider;
+import org.apache.ambari.server.controller.spi.SystemException;
+import org.apache.ambari.server.controller.spi.UnsupportedPropertyException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Sets;
+import com.google.inject.assistedinject.Assisted;
+import com.google.inject.assistedinject.AssistedInject;
+
+/**
+ * Resource provider for auth resources.
+ */
+public class AuthResourceProvider extends AbstractControllerResourceProvider implements ResourcePredicateEvaluator {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AuthResourceProvider.class);
+
+  // ----- Property ID constants ---------------------------------------------
+
+  public static final String AUTH_RESOURCE_CATEGORY = "Auth";
+
+  public static final String USERNAME_PROPERTY_ID = "user_name";
+
+  public static final String AUTH_USERNAME_PROPERTY_ID = AUTH_RESOURCE_CATEGORY + "/" + USERNAME_PROPERTY_ID;
+
+  /**
+   * The key property ids for a User resource.
+   */
+  private static Map<Resource.Type, String> keyPropertyIds = ImmutableMap.<Resource.Type, String>builder()
+      .put(Resource.Type.Auth, AUTH_USERNAME_PROPERTY_ID)
+      .build();
+
+  private static Set<String> propertyIds = Sets.newHashSet(
+      AUTH_USERNAME_PROPERTY_ID
+  );
+
+  /**
+   * Create a new resource provider for the given management controller.
+   */
+  @AssistedInject
+  AuthResourceProvider(@Assisted AmbariManagementController managementController) {
+    super(Resource.Type.Auth, propertyIds, keyPropertyIds, managementController);
+  }
+
+  @Override
+  public RequestStatus createResourcesAuthorized(Request request)
```

Comment:
Can we simplify the current provider: here we will no data and use it as a safe spot to authorize the user with post request.  The user profile could be pulled afterwards using old api endpoint.